pkgs/applications/audio/pulseaudio-module-sndio/default.nix

```
@@ -0,0 +1,41 @@
+{ stdenv, runCommand, fetchFromGitHub, pkgconfig, sndio, pulseaudio, libtool }:
+
+let
+  pulsecore = runCommand "pulseaudio-sources" {} ''
+    tar -xf ${pulseaudio.src}
+    mv pulseaudio*/src/pulsecore $out
+  '';
+in
+  stdenv.mkDerivation rec {
+  pname = "pulseaudio-module-sndio";
+  version = "12.0";
+
+  src = fetchFromGitHub {
+    owner = "t6";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:1xngzqhg01df9fgadv10nmjxpzsykycbxrs241ilpr98x1vf1kn5";
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ sndio pulseaudio libtool ];
+  makeFlags = [
+    "DESTDIR=$$out"
```

```suggestion
    "DESTDIR=${placeholder "out"}"
```
This is probably less hacky.

---

Done. There's a few places in nixpkgs which do it exactly like I did, but the majority follows your suggestion. Maybe those should be refactored.